tests/rules/javascript/test.widget_module.js

```diff
@@ -0,0 +1,61 @@
+import { VALIDATION_WARNING } from 'const';
+import JavaScriptScanner from 'scanners/javascript';
+import * as messages from 'messages';
+
+
+describe('widget_module', () => {
+
+  it('should catch require of sdk/widget as literal', () => {
+    var code = `require("sdk/widget").Widget({
+      id: "mozilla-icon",
+      label: "My Mozilla Widget",
+      contentURL: "http://www.mozilla.org/favicon.ico"
+    });`;
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    return jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].code,
+                     messages.DEPREC_SDK_MOD_WIDGET.code);
+        assert.equal(validationMessages[0].type, VALIDATION_WARNING);
+      });
+  });
+
+  it('should catch require of sdk/widget as var', () => {
+    var code = `var path = 'sdk/widget';
+    require(path).Widget({
+      id: "mozilla-icon",
+      label: "My Mozilla Widget",
+      contentURL: "http://www.mozilla.org/favicon.ico"
+    });`;
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    return jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].code,
+                     messages.DEPREC_SDK_MOD_WIDGET.code);
+        assert.equal(validationMessages[0].type, VALIDATION_WARNING);
+      });
+  });
+
+  it('should catch require() first arg being a global', () => {
```

Comment:
Yep - so looking at the code I don't see a good way to explain this rule further that isn't overlapping with the way the other two tests are working as they're already very explicit. This specific case it doesn't make much sense to put some different test code and assert nothing is caught - because that's too open-ended imho - but maybe I'm misunderstanding something as to how you see we could enhance the story the tests tell?

If you have an idea how to do this feel free to add an issue and we can tackle it separately.